chisel/src/executor.rs

```diff
@@ -176,12 +176,21 @@ impl SessionSource {
             .get(input)
             .or_else(|| source.infer_inner_expr_type());
 
-        let (contract_expr, ty) = match contract_expr
-            .and_then(|e| Type::ethabi(e, Some(&generated_output.intermediate)).map(|ty| (e, ty)))
-        {
-            Some(res) => res,
-            // this type was denied for inspection, continue
-            None => return Ok((true, None)),
+        // If the current action is a function call, we get its return type
+        // otherwise it returns None
+        let function_call_return_type =
+            Type::get_function_return_type(contract_expr, &generated_output.intermediate);
+
+        let (contract_expr, ty) = if function_call_return_type.is_some() {
+            (contract_expr.unwrap(), function_call_return_type.unwrap())
```

Comment:
I think this can be an `if let (Some(), Some())` so we can get rid of the unwraps here